cairo/tests/ethereum/cancun/vm/precompiled_contracts/test_alt_bn128.py

```diff
@@ -26,10 +27,90 @@ def data_strategy(draw):
         return draw(st.binary(min_size=0, max_size=base * 193))
 
 
+@st.composite
+def add_strategy(draw):
+    """
+    Strategy for generating test cases for the alt_bn128_add precompiled contract.
+    Covers:
+    - Valid points on the ALT_BN128 curve (including point at infinity).
+    - Invalid inputs: coordinates >= ALT_BN128_PRIME or points not on the curve.
+    """
+    case_type = draw(
+        st.sampled_from(
+            [
+                "both_valid",
+                "first_infinity",
+                "second_infinity",
+                "out_of_range",
+                "invalid_point",
+            ]
+        )
+    )
+
+    def generate_valid_point():
+        generator_point = BNP(BNF(1), BNF(2))
+        scalar = draw(st.integers(min_value=0, max_value=ALT_BN128_PRIME - 1))
+        point = generator_point.mul_by(scalar)
+        return int(point.x), int(point.y)
```

Comment:
you can `from cairo_ec.curve import AltBn128` and use `AltBn128.random_point()` and `AltBn128.is_on_curve()`